### PR TITLE
[Compute] Create a new extension `image-gallery`

### PR DESCRIPTION
**Pull request by zhoxing-ms**
Feature request: https://github.com/Azure/azure-cli/issues/19882

Related PR: https://github.com/Azure/azure-cli/pull/20129

Add a new extension to support requirements related to image gallery (include community gallery)

*Please note that at present, due to the REST service does not release the features of list operation, so the code related to list operation has been hidden by comments*

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/image-gallery/setup.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+# The full list of classifiers is available at
+# https://pypi.python.org/pypi?%3Aaction=list_classifiers
+CLASSIFIERS = [
+    'Development Status :: 4 - Beta',
+    'Intended Audience :: Developers',
+    'Intended Audience :: System Administrators',
+    'Programming Language :: Python',
+    'Programming Language :: Python :: 3',
+    'Programming Language :: Python :: 3.4',
```
Comment (wangzelin007) [inline src/image-gallery/setup.py:29]: Please update programming language.

File: src/image-gallery/setup.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    'Programming Language :: Python :: 3.5',
+    'Programming Language :: Python :: 3.6',
```
Comment (zhoxing-ms) [inline src/image-gallery/setup.py:31]:
```suggestion
    'Programming Language :: Python :: 3',
    'Programming Language :: Python :: 3.6',
    'Programming Language :: Python :: 3.7',
    'Programming Language :: Python :: 3.8',
```

File: src/image-gallery/azext_image_gallery/_help.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+long-summary: Get a gallery that has been community (private preview feature, please contact community image gallery team by email sigpmdev@microsoft.com to register for preview if you're interested in using this feature).
+examples:
+  - name: Get a gallery that has been community in the given location.
+    text: |
+        az sig show-community --public-gallery-name publicGalleryName --location myLocation
+"""
+
+helps['sig image-definition show-community'] = """
+type: command
+short-summary: Get a community gallery image (preview).
```
Comment (zhoxing-ms) [inline src/image-gallery/azext_image_gallery/_help.py:24]:
```suggestion
short-summary: Get an image in a gallery community (preview).
```

Comment (wangzelin007) [inline src/image-gallery/azext_image_gallery/_help.py]: Better to unify help sentence structure.

Comment (wangzelin007) [inline src/image-gallery/azext_image_gallery/_help.py]: Better to unify help sentence structure.

**Review by wangzelin007 — APPROVED** [review]
LGTM

File: src/image-gallery/azext_image_gallery/_help.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+examples:
+  - name: Get a gallery that has been community in the given location.
+    text: |
+        az sig show-community --public-gallery-name publicGalleryName --location myLocation
+"""
+
+helps['sig image-definition show-community'] = """
+type: command
+short-summary: Get an image in a gallery community (preview).
+long-summary: Get a community gallery image that has been community (private preview feature, please contact community image gallery team by email sigpmdev@microsoft.com to register for preview if you're interested in using this feature).
```
Comment (zhoxing-ms) [inline src/image-gallery/azext_image_gallery/_help.py:25]:
```suggestion
long-summary: Get an image in a gallery community (private preview feature, please contact community image gallery team by email sigpmdev@microsoft.com to register for preview if you're interested in using this feature).
```

**Comment by yonzhan** [comment]
Compute

**Pull request merged**